src/ribbon/buttonbar.cpp

```diff
@@ -130,15 +112,25 @@ class wxRibbonButtonBarButtonBase
 
         if ( bitmap_large.IsOk() )
         {
+            // wxImageList handles logical size inconsistent between some platforms,
+            // which has to be kept for compatibility reasons.
+            // We use physical size for storing and ensure, that logical size is the same.
+            bitmap_large.SetScaleFactor(1.0);
```

Comment:
Sorry but does this really work under GTK/macOS, i.e. platforms where the scale factor is actually used when drawing the bitmap? If you have already tested it there and it does work, I must be missing something, but if not, I'm afraid this isn't the right thing to do.